hdump.c

```diff
@@ -116,12 +114,15 @@ int main(int argc, char *argv[])
 			/* imprime os bytes separados por espaço */
 			printf("%02x%*c", (unsigned int) *(buff+i), (i+1 == cols/2) ? 2 : 1, ' ');
 
-			/* define o fim do array ascii (sera usado como string) */
-			*(ascii+bread) = '\0';
-
-			/* imprime os caracteres ascii */
-			if (i == bread-1)
```

Comment:
I moved the ascii assignement to define its end to inside the if condition once that this assignment might happen only after every bread gets printed, this way we can avoid unnecessary assignment into ascii every time the loop restart.